pkg/model/provider/openai/client.go

```diff
@@ -148,6 +148,11 @@ func convertMessages(messages []chat.Message) []openai.ChatCompletionMessage {
 			continue
 		}
 
+		// Handle an empty tool message. This can happen if the tool produced no output.
```

Comment:
This could be done once for all the providers if we handle it in the mcp client: https://github.com/docker/cagent/blob/main/pkg/tools/mcp/client.go#L199-L210

Reply:
Thanks, that makes more sense. Tested it and works fine! 

<img width="1389" height="462" alt="image" src="https://github.com/user-attachments/assets/b0142a0e-db77-4292-8833-78ec5d65fa26" />